Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/event/dao/DatabaseEventLoggerDao.java

```diff
@@ -69,220 +74,126 @@ private SingletonHolder() {
 
-        try {
-            for (DatabaseEvent event : events) {
-                JSONObject jObject = new JSONObject(event);
-                String exceptionClass = jObject.getString("exceptionClass");
-                if (StringUtils.isNotBlank(exceptionClass)) {
-                    exceptions.add(exceptionClass.replaceAll("class", ""));
-                }
-            }
-        } catch (JSONException e) {
-            LOG.error("Exception getting Exceptions messages caused by :{}", e.getLocalizedMessage(), e);
+    public Set<String> getExceptions() {
+        Set<String> exceptions = new HashSet<>();
+        for (EventDTO event : getAllFailureMessages(null, null, null, null)) {
+            exceptions.add(event.getExceptionType());
         }
         return exceptions;
     }
 
-    /**
-     * Get all failure events for certain criteria
-     *
-     * @param serviceType
-     * @param startDate
-     * @param endDate
-     * @return
-     */
-    public List<DatabaseEvent> getAllFailureMessages(String serviceType, Date startDate, Date endDate) {
-        Session session = null;
-        List<DatabaseEvent> events = null;
 
-        try {
-            final SessionFactory sessionFactory = getSessionFactory();
-            if (sessionFactory != null) {
-                session = sessionFactory.openSession();
-                Criteria criteria = session.createCriteria(DatabaseEvent.class);
+    public List<EventDTO> getAllFailureMessages(String serviceType, String exceptionType, Date startDate,
+        Date endDate) {
+        return filterEventsByException(
+            findAllBy(getSession(), DatabaseEvent.class, buildCriterionFor(serviceType, startDate, endDate)),
+            exceptionType);
+    }
 
-                criteria.add(Restrictions.eq(EVENT_TYPE_NAME, "MESSAGE_PROCESSING_FAILED"));
+    protected SessionFactory getSessionFactory() {
+        SessionFactory fact = null;
+        HibernateUtil util = HibernateUtilFactory.getEventHibernateUtil();
+        if (util != null) {
+            fact = util.getSessionFactory();
+        }
+        return fact;
+    }
 
-                if (startDate != null) {
-                    criteria.add(Restrictions.ge(DATE_NAME, startDate));
-                }
+    protected Session getSession() {
+        return getSessionFactory().openSession();
+    }
 
-                if (endDate != null) {
-                    criteria.add(Restrictions.le(DATE_NAME, endDate));
-                }
+    private static List<EventDTO> filterEventsByException(List<DatabaseEvent> events, String exceptionType) {
+        LinkedList<EventDTO> list = new LinkedList<>();
 
-                if (StringUtils.isNotBlank(serviceType)) {
-                    criteria.add(Restrictions.eq(SERVICE_TYPE, serviceType));
+        for (DatabaseEvent event : events) {
+            EventDTO dto = EventDTO.convertFrom(event, OMIT_DESCRIPTION);
+            if (StringUtils.isNotBlank(exceptionType)) {
```

Comment:
Optional: can combine null check for both with apache common StringUtils.equals()

Reply:
for the logic of a filted-list we cannot combine the two;
if the exceptionType is blank return the all-list; if the type is not-blank and equals return a filted-list